src/main/java/org/jitsi/jibri/JibriManager.kt

```diff
@@ -0,0 +1,203 @@
+    val youTubeStreamKey: String
+)
+
+/**
+ * [JibriManager] is responsible for managing the various services Jibri
+ * provides, as well as providing an API to query the health of this Jibri
+ * instance.  NOTE: currently Jibri only runs a single service at a time, so
+ * if one is running, the Jibri will describe itself as busy
+ */
+class JibriManager(private val config: JibriConfig) : StatusPublisher<JibriStatusPacketExt.Status>() {
+    private val logger = Logger.getLogger(this::class.qualifiedName)
+    private var currentActiveService: JibriService? = null
+    private var pendingIdleFunc: () -> Unit = {}
+    val executor = Executors.newSingleThreadScheduledExecutor(NameableThreadFactory("JibriManager"))
+    var serviceTimeoutTask: ScheduledFuture<*>? = null
+
+    /**
+     * Starts a [FileRecordingJibriService] to record the call described
+     * in the params to a file.  Returns a [StartServiceResult] to denote
+     * whether the service was started successfully or not.
+     */
+    @Synchronized
+    fun startFileRecording(
+            serviceParams: ServiceParams,
+            fileRecordingParams: FileRecordingParams,
+            serviceStatusHandler: JibriServiceStatusHandler? = null): StartServiceResult {
+        logger.info("Starting a file recording with params: $fileRecordingParams")
+        if (busy()) {
+            logger.info("Jibri is busy, can't start service")
+            return StartServiceResult.BUSY
+        }
+        val service = FileRecordingJibriService(
+                RecordingOptions(
+                    config.recordingDirectory,
+                    fileRecordingParams.callParams,
+                    config.finalizeRecordingScriptPath
+                )
+            )
+        return startService(service, serviceParams, serviceStatusHandler)
+    }
+
+    /**
+     * Starts a [StreamingJibriService] to capture the call according
+     * to [streamingParams].  Returns a [StartServiceResult] to
+     * denote whether the service was started successfully or not.
+     */
+    @Synchronized
+    fun startStreaming(
+            serviceParams: ServiceParams,
+            streamingParams: StreamingParams,
+            serviceStatusHandler: JibriServiceStatusHandler? = null): StartServiceResult {
+        logger.info("Starting a stream with params: $streamingParams")
+        if (busy()) {
+            logger.info("Jibri is busy, can't start service")
+            return StartServiceResult.BUSY
+        }
+        val service = StreamingJibriService(StreamingOptions(
+                streamingParams.youTubeStreamKey,
+                streamingParams.callParams
+        ))
+        return startService(service, serviceParams, serviceStatusHandler)
+    }
+
+    /**
+     * Helper method to handle the boilerplate of starting a [JibriService].
+     * Returns a [StartServiceResult] to denote whether the service was
+     * started successfully or not.
+     */
+    private fun startService(
```

Comment:
Pardon the ignorance, what stops multiple places from calling startService at the same time? If that's possible, is that a problem?

Reply:
this is a private function and it's only called from methods which are all marked `@Synchronized`